### PR TITLE
http or s

### PR DESCRIPTION
**Pull request by bloodearnest**
- fix justfile warning
- fix missing f string
- Refactor url generation to better handle http or https

### CODE REVIEW

File: hatch/client.py [hunk 1]
```diff
@@ -35,7 +35,12 @@ def fetch_index(token, workspace, release_id=None):
         kwargs["release_id"] = release_id
     else:
         view_name = "workspace_index"
-    resp = requests.get(app.reverse_url(view_name, **kwargs), headers=headers)
+
+    # generate url
+    path = app.app.url_path_for(view_name, **kwargs)
```
Comment (madwort) [inline hatch/client.py:40]: `app.app.url_path_for` looks like a typo is that correct??

Comment (madwort) [reply hatch/client.py:40]: (this may be a stupid question...)

Comment (bloodearnest) [reply hatch/client.py:40]: So, `hatch.app`  is a module, and `hatch.app.app` is an instance of a FastAPI app.

Not great naming

Comment (madwort) [reply hatch/client.py:40]: ok then

**Pull request merged**